crates/core_simd/src/round.rs

```diff
@@ -19,6 +19,11 @@ macro_rules! implement {
             /// * Not be NaN
             /// * Not be infinite
             /// * Be representable in the return type, after truncating off its fractional part
+            ///
+            /// If these requirements are infeasible or costly, consider using the safe function [cast],
```

Comment:
I think I would word it the other way really, prefer `cast` unless for some reason you absolutely know these are all true.